File: Dataset/OOP_prepared/Java/Circle_trace_1.java
public class Circle {
    private double radius;

    public Circle() {
        radius = 1.0;
    }

    public Circle(double radius) {
        this.radius = radius;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }
    
    public double getArea() {
      return radius*radius*Math.PI;
    }

    public double getCircumference(){
        return radius*2*Math.PI;
    }

    @Override
    public String toString() {
        return "Circle[" + "radius=" + radius + ']';
    }
}

class TestCircle{
    public static void main(String[] args) {
        Circle c = new Circle(3.5);
        System.out.printf("area is: %.2f%n", c.getArea());
    }
}

File: Dataset/OOP_prepared/Python/Circle_trace_1.py
import math

class Circle:
    def __init__(self, radius=1.0):
        self._radius = radius

    @property
    def radius(self):
        return self._radius

    @radius.setter
    def radius(self, radius):
        self._radius = radius

    def get_area(self):
        return self._radius * self._radius * math.pi

    def get_circumference(self):
        return self._radius * 2 * math.pi

    def __str__(self):
        return f"Circle[radius={self._radius}]"

if __name__ == "__main__":
    c = Circle(3.5)
    print(f"area is: {c.get_area():.2f}")